### PR TITLE
Feature/survey progress

### PR DESCRIPTION
**Pull request by MaxWojtkow**
- refactored Report storage and generation

### CODE REVIEW

File: Backend/src/controllers/SurveyController.ts [hunk 1]
```diff
@@ -730,110 +785,119 @@ export class SurveyController {
...
+
+  static async createScoringReport(surveyId: number, budget: number) {
+    let minBuckets = 4;
+    let cutToZero = true; // if true, privatize returns zero when output is negative, else absolute value
+    let pController = new ParticipantController();
+    let participants = await pController.getParticipants(surveyId);
+
+    const dataset = newArrayView(participants.participants);
+
+    let [min, max, a, b] = [
```
Comment (j0-s3f) [inline Backend/src/controllers/SurveyController.ts:834]: Please describe a and b.
You can hardly understand what this variables do.

File: Backend/src/controllers/SurveyController.ts [hunk 2]
```diff
@@ -730,110 +785,119 @@ export class SurveyController {
...
+      b += width;
     }
-    await SurveyController.updateSurveyPrivacyBudget(
-      body.surveyId,
-      body.privacyBudget - cost
-    );
-    return resultPrivate;
+    return tempPrScoringReport;
+  }
+  static async createResponseTimeReport(/*surveyId: number, budget: number*/) {
```
Comment (j0-s3f) [inline Backend/src/controllers/SurveyController.ts:897]: Please add return type.

Report is a type in the entities and in the CreateReportResponseDto.

Maybe rename the one in the Dto to reference it correctly

File: Backend/src/controllers/SurveyController.ts [hunk 3]
```diff
@@ -730,110 +785,119 @@ export class SurveyController {
...
+    await SurveyController.updateSurveyPrivacyBudget(survey.id, 0);
+
+    await SurveyController.saveReports(
+      survey.id,
+      resultPrivate.scoringReport.histogramData,
+      resultPrivate.responseTimeReport.histogramData
+    );
+  }
+
+  static async createScoringReport(surveyId: number, budget: number) {
```
Comment (j0-s3f) [inline Backend/src/controllers/SurveyController.ts:826]: Please add the return type.

File: Backend/src/controllers/SurveyController.ts [hunk 4]
```diff
@@ -708,7 +711,59 @@ export class SurveyController {
...
+      .then(() => {
+        result.hasError = false;
+      })
+      .catch((e) => {
+        result.hasError = true;
+        result.message = e;
+      });
+  }
+
+  async getReports(body: CloseSurveyDto): Promise<CreateReportResponseDto> {
```
Comment (j0-s3f) [inline Backend/src/controllers/SurveyController.ts:740]: Please create new Dtos specific for this endpoint.

File: Frontend/src/Components/TrusteeLogin/TrusteeLoginContent/Reports/DisplayReport/DisplayReport.tsx [hunk 5]
```diff
@@ -8,19 +8,79 @@ import {
...
+    }/${props.surveyProgress.total} (${(
+      (props.surveyProgress.finished * 100) /
+      props.surveyProgress.total
+    ).toPrecision(3)}%)`;
+    csvExporter.options.filename = `${props.surveyTitle}_Response_Time_Report`;
+    csvExporter.options.headers = ["", ""];
+    csvExporter.generateCsv(JSON.parse(JSON.stringify(data)));
+  };
+
+  const reportChart = (data: HistogramData[]) => (
```
Comment (j0-s3f) [inline Frontend/src/Components/TrusteeLogin/TrusteeLoginContent/Reports/DisplayReport/DisplayReport.tsx:76]: Why not move the Chart to a different file as previously done.

**Review by j0-s3f — CHANGES_REQUESTED** [review]

File: Frontend/src/Components/TrusteeLogin/TrusteeLoginContent/Reports/DisplayReport/DisplayReport.tsx (hunk 5 above)
Comment (MaxWojtkow) [reply Frontend/src/Components/TrusteeLogin/TrusteeLoginContent/Reports/DisplayReport/DisplayReport.tsx:76]: can still be done afterwards if necessary

**Review by j0-s3f — APPROVED** [review]

**Pull request merged**